### ISSUE TITLE
[jsartoolkitNFT] Removing throw warnings while compiling libar.bc

### ISSUE THREAD

**Issue by kalwalt** [webarkit/WebARKitLib#11]
While compiling jsartoolkitNFT we recieve these warnings:

```
In file included from /mnt/c/Users/perda/OneDrive/Documenti/kalwalt-github/webarkit-jsfeat-cpp/emscripten/WebARKitLib/lib/SRC/KPM/FreakMatcher/facade/visual_database_facade.cpp:37:
In file included from /mnt/c/Users/perda/OneDrive/Documenti/kalwalt-github/webarkit-jsfeat-cpp/emscripten/WebARKitLib/lib/SRC/KPM/FreakMatcher/matchers/visual_database-inline.h:44:
/mnt/c/Users/perda/OneDrive/Documenti/kalwalt-github/webarkit-jsfeat-cpp/emscripten/WebARKitLib/lib/SRC/KPM/FreakMatcher/matchers/visual_database.h:83:14: warning: dynamic exception specifications with types are currently ignored in wasm [-Wwasm-exception-spec]
        void addImage(const Image& image, id_t id) throw(Exception);
             ^                                     ~~~~~~~~~~~~~~~~
/mnt/c/Users/perda/OneDrive/Documenti/kalwalt-github/webarkit-jsfeat-cpp/emscripten/WebARKitLib/lib/SRC/KPM/FreakMatcher/matchers/visual_database.h:88:14: warning: dynamic exception specifications with types are currently ignored in wasm [-Wwasm-exception-spec]
        void addImage(const GaussianScaleSpacePyramid* pyramid, id_t id) throw(Exception);
             ^                                                           ~~~~~~~~~~~~~~~~
/mnt/c/Users/perda/OneDrive/Documenti/kalwalt-github/webarkit-jsfeat-cpp/emscripten/WebARKitLib/lib/SRC/KPM/FreakMatcher/matchers/visual_database.h:93:14: warning: dynamic exception specifications with types are currently ignored in wasm [-Wwasm-exception-spec]
        void addKeyframe(keyframe_ptr_t keyframe , id_t id) throw(Exception);
             ^                                              ~~~~~~~~~~~~~~~~
/mnt/c/Users/perda/OneDrive/Documenti/kalwalt-github/webarkit-jsfeat-cpp/emscripten/WebARKitLib/lib/SRC/KPM/FreakMatcher/matchers/visual_database.h:98:14: warning: dynamic exception specifications with types are currently ignored in wasm [-Wwasm-exception-spec]
        bool query(const Image& image) throw(Exception);
             ^                         ~~~~~~~~~~~~~~~~
/mnt/c/Users/perda/OneDrive/Documenti/kalwalt-github/webarkit-jsfeat-cpp/emscripten/WebARKitLib/lib/SRC/KPM/FreakMatcher/matchers/visual_database.h:103:14: warning: dynamic exception specifications with types are currently ignored in wasm [-Wwasm-exception-spec]
        bool query(const GaussianScaleSpacePyramid* pyramid) throw(Exception);
             ^                                               ~~~~~~~~~~~~~~~~
/mnt/c/Users/perda/OneDrive/Documenti/kalwalt-github/webarkit-jsfeat-cpp/emscripten/WebARKitLib/lib/SRC/KPM/FreakMatcher/matchers/visual_database.h:104:14: warning: dynamic exception specifications with types are currently ignored in wasm [-Wwasm-exception-spec]
        bool query(const keyframe_t* query_keyframe) throw(Exception);
             ^                                       ~~~~~~~~~~~~~~~~
6 warnings generated.
```
The code will  be compiled with emsdk 3.1.20 but warnings will be converted as errors with emsdk 3.1.26, this because emscripten will use c++ 17 by default, dynamic throwing is not allowed anymore.
We need to remove that lines, and/or if we have time and better throw exceptions. But this will require more time to develop.

**Comment by kalwalt** [comment]
note: it's also required to remove throw's from the implementations otherwise you got errors like these:

```
In file included from /mnt/c/Users/perda/OneDrive/Documenti/kalwalt-github/webarkit-org/jsartoolkitNFT/emscripten/WebARKitLib/lib/SRC/KPM/FreakMatcher/facade/visual_database_facade.cpp:37:
/mnt/c/Users/perda/OneDrive/Documenti/kalwalt-github/webarkit-org/jsartoolkitNFT/emscripten/WebARKitLib/lib/SRC/KPM/FreakMatcher/matchers/visual_database-inline.h:79:61: error: exception specification in declaration does not match previous declaration
    void VisualDatabase<FEATURE_EXTRACTOR, STORE, MATCHER>::addImage(const vision::Image& image, id_t id) throw(Exception) {
                                                            ^
/mnt/c/Users/perda/OneDrive/Documenti/kalwalt-github/webarkit-org/jsartoolkitNFT/emscripten/WebARKitLib/lib/SRC/KPM/FreakMatcher/matchers/visual_database.h:83:14: note: previous declaration is here
        void addImage(const Image& image, id_t id);
             ^
In file included from /mnt/c/Users/perda/OneDrive/Documenti/kalwalt-github/webarkit-org/jsartoolkitNFT/emscripten/WebARKitLib/lib/SRC/KPM/FreakMatcher/facade/visual_database_facade.cpp:37:
/mnt/c/Users/perda/OneDrive/Documenti/kalwalt-github/webarkit-org/jsartoolkitNFT/emscripten/WebARKitLib/lib/SRC/KPM/FreakMatcher/matchers/visual_database-inline.h:102:61: error: exception specification in declaration does not match previous declaration
    void VisualDatabase<FEATURE_EXTRACTOR, STORE, MATCHER>::addImage(const GaussianScaleSpacePyramid* pyramid, id_t id) throw(Exception) {
                                                            ^
/mnt/c/Users/perda/OneDrive/Documenti/kalwalt-github/webarkit-org/jsartoolkitNFT/emscripten/WebARKitLib/lib/SRC/KPM/FreakMatcher/matchers/visual_database.h:88:14: note: previous declaration is here
        void addImage(const GaussianScaleSpacePyramid* pyramid, id_t id);
             ^
In file included from /mnt/c/Users/perda/OneDrive/Documenti/kalwalt-github/webarkit-org/jsartoolkitNFT/emscripten/WebARKitLib/lib/SRC/KPM/FreakMatcher/facade/visual_database_facade.cpp:37:
/mnt/c/Users/perda/OneDrive/Documenti/kalwalt-github/webarkit-org/jsartoolkitNFT/emscripten/WebARKitLib/lib/SRC/KPM/FreakMatcher/matchers/visual_database-inline.h:132:61: error: exception specification in declaration does not match previous declaration
    void VisualDatabase<FEATURE_EXTRACTOR, STORE, MATCHER>::addKeyframe(keyframe_ptr_t keyframe , id_t id) throw(Exception) {
                                                            ^
/mnt/c/Users/perda/OneDrive/Documenti/kalwalt-github/webarkit-org/jsartoolkitNFT/emscripten/WebARKitLib/lib/SRC/KPM/FreakMatcher/matchers/visual_database.h:93:14: note: previous declaration is here
        void addKeyframe(keyframe_ptr_t keyframe , id_t id);
             ^
In file included from /mnt/c/Users/perda/OneDrive/Documenti/kalwalt-github/webarkit-org/jsartoolkitNFT/emscripten/WebARKitLib/lib/SRC/KPM/FreakMatcher/facade/visual_database_facade.cpp:37:
/mnt/c/Users/perda/OneDrive/Documenti/kalwalt-github/webarkit-org/jsartoolkitNFT/emscripten/WebARKitLib/lib/SRC/KPM/FreakMatcher/matchers/visual_database-inline.h:142:61: error: exception specification in declaration does not match previous declaration
    bool VisualDatabase<FEATURE_EXTRACTOR, STORE, MATCHER>::query(const vision::Image& image) throw(Exception) {
                                                            ^
/mnt/c/Users/perda/OneDrive/Documenti/kalwalt-github/webarkit-org/jsartoolkitNFT/emscripten/WebARKitLib/lib/SRC/KPM/FreakMatcher/matchers/visual_database.h:98:14: note: previous declaration is here
        bool query(const Image& image);
             ^
In file included from /mnt/c/Users/perda/OneDrive/Documenti/kalwalt-github/webarkit-org/jsartoolkitNFT/emscripten/WebARKitLib/lib/SRC/KPM/FreakMatcher/facade/visual_database_facade.cpp:37:
/mnt/c/Users/perda/OneDrive/Documenti/kalwalt-github/webarkit-org/jsartoolkitNFT/emscripten/WebARKitLib/lib/SRC/KPM/FreakMatcher/matchers/visual_database-inline.h:160:61: error: exception specification in declaration does not match previous declaration
    bool VisualDatabase<FEATURE_EXTRACTOR, STORE, MATCHER>::query(const GaussianScaleSpacePyramid* pyramid) throw(Exception) {
                                                            ^
/mnt/c/Users/perda/OneDrive/Documenti/kalwalt-github/webarkit-org/jsartoolkitNFT/emscripten/WebARKitLib/lib/SRC/KPM/FreakMatcher/matchers/visual_database.h:103:14: note: previous declaration is here
        bool query(const GaussianScaleSpacePyramid* pyramid);
             ^
In file included from /mnt/c/Users/perda/OneDrive/Documenti/kalwalt-github/webarkit-org/jsartoolkitNFT/emscripten/WebARKitLib/lib/SRC/KPM/FreakMatcher/facade/visual_database_facade.cpp:37:
/mnt/c/Users/perda/OneDrive/Documenti/kalwalt-github/webarkit-org/jsartoolkitNFT/emscripten/WebARKitLib/lib/SRC/KPM/FreakMatcher/matchers/visual_database-inline.h:180:61: error: exception specification in declaration does not match previous declaration
    bool VisualDatabase<FEATURE_EXTRACTOR, STORE, MATCHER>::query(const keyframe_t* query_keyframe) throw(Exception) {
                                                            ^
/mnt/c/Users/perda/OneDrive/Documenti/kalwalt-github/webarkit-org/jsartoolkitNFT/emscripten/WebARKitLib/lib/SRC/KPM/FreakMatcher/matchers/visual_database.h:104:14: note: previous declaration is here
        bool query(const keyframe_t* query_keyframe);
             ^
6 errors generated.
```

**Comment by kalwalt** [comment]
The above commits are not enough see this errors from [github actions](https://github.com/webarkit/jsartoolkitNFT/actions/runs/3569164324/jobs/5998861074):
```
/src/emscripten/WebARKitLib/lib/SRC/KPM/FreakMatcher/framework/image.h:89:37: error: ISO C++17 does not allow dynamic exception specifications [-Wdynamic-exception-spec]
                   size_t channels) throw(Exception);
                                    ^~~~~~~~~~~~~~~~
/src/emscripten/WebARKitLib/lib/SRC/KPM/FreakMatcher/framework/image.h:89:37: note: use 'noexcept(false)' instead
                   size_t channels) throw(Exception);
                                    ^~~~~~~~~~~~~~~~
                                    noexcept(false)
1 error generated.
```

**Comment by kalwalt** [comment]
there are dynamic throw everywhere:

```
/src/emscripten/WebARKitLib/lib/SRC/KPM/FreakMatcher/framework/image.h:89:37: error: ISO C++17 does not allow dynamic exception specifications [-Wdynamic-exception-spec]
                   size_t channels) throw(Exception);
                                    ^~~~~~~~~~~~~~~~
/src/emscripten/WebARKitLib/lib/SRC/KPM/FreakMatcher/framework/image.h:89:37: note: use 'noexcept(false)' instead
                   size_t channels) throw(Exception);
                                    ^~~~~~~~~~~~~~~~
                                    noexcept(false)
1 error generated.
```
https://github.com/webarkit/jsartoolkitNFT/actions/runs/3569326336/jobs/5999169640

**Comment by kalwalt** [comment]
I can see taht there are other throw in the WebARKitLib code, i will post here a list tomorrow or when i have time.

**Comment by kalwalt** [comment]
The actual Eigen lib, included in KPM, can not be compiled with emsdk 3.1.26, see the error:
```
In file included from C:\Users\perda\OneDrive\Documenti\kalwalt-github\webarkit-org\jsartoolkitNFT\emscripten\WebARKitLib\lib\SRC\KPM\FreakMatcher\facade\visual_database_facade.cpp:37:
In file included from C:\Users\perda\OneDrive\Documenti\kalwalt-github\webarkit-org\jsartoolkitNFT\emscripten\WebARKitLib\lib\SRC\KPM\FreakMatcher\matchers/visual_database-inline.h:44:
In file included from C:\Users\perda\OneDrive\Documenti\kalwalt-github\webarkit-org\jsartoolkitNFT\emscripten\WebARKitLib\lib\SRC\KPM\FreakMatcher\matchers/visual_database.h:44:
In file included from C:\Users\perda\OneDrive\Documenti\kalwalt-github\webarkit-org\jsartoolkitNFT\emscripten\WebARKitLib\lib\SRC\KPM\FreakMatcher\homography_estimation/robust_homography.h:45:
In file included from C:\Users\perda\OneDrive\Documenti\kalwalt-github\webarkit-org\jsartoolkitNFT\emscripten\WebARKitLib\lib\SRC\KPM\FreakMatcher\Eigen/Core:275:
C:\Users\perda\OneDrive\Documenti\kalwalt-github\webarkit-org\jsartoolkitNFT\emscripten\WebARKitLib\lib\SRC\KPM\FreakMatcher\Eigen/src/Core/Functors.h:891:28: error: no member named 'binder2nd' in namespace 'std'
struct functor_traits<std::binder2nd<T> >
                      ~~~~~^
C:\Users\perda\OneDrive\Documenti\kalwalt-github\webarkit-org\jsartoolkitNFT\emscripten\WebARKitLib\lib\SRC\KPM\FreakMatcher\Eigen/src/Core/Functors.h:891:38: error: 'T' does not refer to a value
struct functor_traits<std::binder2nd<T> >
                                     ^
C:\Users\perda\OneDrive\Documenti\kalwalt-github\webarkit-org\jsartoolkitNFT\emscripten\WebARKitLib\lib\SRC\KPM\FreakMatcher\Eigen/src/Core/Functors.h:890:19: note: declared here
template<typename T>
                  ^
C:\Users\perda\OneDrive\Documenti\kalwalt-github\webarkit-org\jsartoolkitNFT\emscripten\WebARKitLib\lib\SRC\KPM\FreakMatcher\Eigen/src/Core/Functors.h:891:41: error: expected unqualified-id
struct functor_traits<std::binder2nd<T> >
                                        ^
C:\Users\perda\OneDrive\Documenti\kalwalt-github\webarkit-org\jsartoolkitNFT\emscripten\WebARKitLib\lib\SRC\KPM\FreakMatcher\Eigen/src/Core/Functors.h:895:28: error: no member named 'binder1st' in namespace 'std'
struct functor_traits<std::binder1st<T> >
                      ~~~~~^
C:\Users\perda\OneDrive\Documenti\kalwalt-github\webarkit-org\jsartoolkitNFT\emscripten\WebARKitLib\lib\SRC\KPM\FreakMatcher\Eigen/src/Core/Functors.h:895:38: error: 'T' does not refer to a value
struct functor_traits<std::binder1st<T> >
                                     ^
C:\Users\perda\OneDrive\Documenti\kalwalt-github\webarkit-org\jsartoolkitNFT\emscripten\WebARKitLib\lib\SRC\KPM\FreakMatcher\Eigen/src/Core/Functors.h:894:19: note: declared here
template<typename T>
                  ^
C:\Users\perda\OneDrive\Documenti\kalwalt-github\webarkit-org\jsartoolkitNFT\emscripten\WebARKitLib\lib\SRC\KPM\FreakMatcher\Eigen/src/Core/Functors.h:895:41: error: expected unqualified-id
struct functor_traits<std::binder1st<T> >
                                        ^
In file included from C:\Users\perda\OneDrive\Documenti\kalwalt-github\webarkit-org\jsartoolkitNFT\emscripten\WebARKitLib\lib\SRC\KPM\FreakMatcher\facade\visual_database_facade.cpp:37:
In file included from C:\Users\perda\OneDrive\Documenti\kalwalt-github\webarkit-org\jsartoolkitNFT\emscripten\WebARKitLib\lib\SRC\KPM\FreakMatcher\matchers/visual_database-inline.h:44:
In file included from C:\Users\perda\OneDrive\Documenti\kalwalt-github\webarkit-org\jsartoolkitNFT\emscripten\WebARKitLib\lib\SRC\KPM\FreakMatcher\matchers/visual_database.h:44:
In file included from C:\Users\perda\OneDrive\Documenti\kalwalt-github\webarkit-org\jsartoolkitNFT\emscripten\WebARKitLib\lib\SRC\KPM\FreakMatcher\homography_estimation/robust_homography.h:45:
In file included from C:\Users\perda\OneDrive\Documenti\kalwalt-github\webarkit-org\jsartoolkitNFT\emscripten\WebARKitLib\lib\SRC\KPM\FreakMatcher\Eigen/Core:278:
In file included from C:\Users\perda\OneDrive\Documenti\kalwalt-github\webarkit-org\jsartoolkitNFT\emscripten\WebARKitLib\lib\SRC\KPM\FreakMatcher\Eigen/src/Core/MatrixBase.h:151:
C:\Users\perda\OneDrive\Documenti\kalwalt-github\webarkit-org\jsartoolkitNFT\emscripten\WebARKitLib\lib\SRC\KPM\FreakMatcher\Eigen/src/Core/../plugins/MatrixCwiseUnaryOps.h:77:32: error: no member named 'binder1st' in namespace 'std'
inline const CwiseUnaryOp<std::binder1st<std::equal_to<Scalar> >, const Derived>
                          ~~~~~^
C:\Users\perda\OneDrive\Documenti\kalwalt-github\webarkit-org\jsartoolkitNFT\emscripten\WebARKitLib\lib\SRC\KPM\FreakMatcher\Eigen/src/Core/../plugins/MatrixCwiseUnaryOps.h:77:64: error: expected '(' for function-style cast or type construction
inline const CwiseUnaryOp<std::binder1st<std::equal_to<Scalar> >, const Derived>
                                         ~~~~~~~~~~~~~~~~~~~~~ ^
C:\Users\perda\OneDrive\Documenti\kalwalt-github\webarkit-org\jsartoolkitNFT\emscripten\WebARKitLib\lib\SRC\KPM\FreakMatcher\Eigen/src/Core/../plugins/MatrixCwiseUnaryOps.h:77:65: error: expected member name or ';' after declaration specifiers
inline const CwiseUnaryOp<std::binder1st<std::equal_to<Scalar> >, const Derived>
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
In file included from C:\Users\perda\OneDrive\Documenti\kalwalt-github\webarkit-org\jsartoolkitNFT\emscripten\WebARKitLib\lib\SRC\KPM\FreakMatcher\facade\visual_database_facade.cpp:37:
In file included from C:\Users\perda\OneDrive\Documenti\kalwalt-github\webarkit-org\jsartoolkitNFT\emscripten\WebARKitLib\lib\SRC\KPM\FreakMatcher\matchers/visual_database-inline.h:44:
In file included from C:\Users\perda\OneDrive\Documenti\kalwalt-github\webarkit-org\jsartoolkitNFT\emscripten\WebARKitLib\lib\SRC\KPM\FreakMatcher\matchers/visual_database.h:44:
In file included from C:\Users\perda\OneDrive\Documenti\kalwalt-github\webarkit-org\jsartoolkitNFT\emscripten\WebARKitLib\lib\SRC\KPM\FreakMatcher\homography_estimation/robust_homography.h:45:
In file included from C:\Users\perda\OneDrive\Documenti\kalwalt-github\webarkit-org\jsartoolkitNFT\emscripten\WebARKitLib\lib\SRC\KPM\FreakMatcher\Eigen/Core:283:
C:\Users\perda\OneDrive\Documenti\kalwalt-github\webarkit-org\jsartoolkitNFT\emscripten\WebARKitLib\lib\SRC\KPM\FreakMatcher\Eigen/src/Core/Assign.h:565:51: error: out-of-line definition of 'operator=' does not match any declaration in 'MatrixBase<Derived>'
EIGEN_STRONG_INLINE Derived& MatrixBase<Derived>::operator=(const MatrixBase& other)
                                                  ^~~~~~~~
In file included from C:\Users\perda\OneDrive\Documenti\kalwalt-github\webarkit-org\jsartoolkitNFT\emscripten\WebARKitLib\lib\SRC\KPM\FreakMatcher\facade\visual_database_facade.cpp:37:
In file included from C:\Users\perda\OneDrive\Documenti\kalwalt-github\webarkit-org\jsartoolkitNFT\emscripten\WebARKitLib\lib\SRC\KPM\FreakMatcher\matchers/visual_database-inline.h:44:
In file included from C:\Users\perda\OneDrive\Documenti\kalwalt-github\webarkit-org\jsartoolkitNFT\emscripten\WebARKitLib\lib\SRC\KPM\FreakMatcher\matchers/visual_database.h:44:
In file included from C:\Users\perda\OneDrive\Documenti\kalwalt-github\webarkit-org\jsartoolkitNFT\emscripten\WebARKitLib\lib\SRC\KPM\FreakMatcher\homography_estimation/robust_homography.h:45:
In file included from C:\Users\perda\OneDrive\Documenti\kalwalt-github\webarkit-org\jsartoolkitNFT\emscripten\WebARKitLib\lib\SRC\KPM\FreakMatcher\Eigen/Core:347:
In file included from C:\Users\perda\OneDrive\Documenti\kalwalt-github\webarkit-org\jsartoolkitNFT\emscripten\WebARKitLib\lib\SRC\KPM\FreakMatcher\Eigen/src/Core/ArrayBase.h:121:
C:\Users\perda\OneDrive\Documenti\kalwalt-github\webarkit-org\jsartoolkitNFT\emscripten\WebARKitLib\lib\SRC\KPM\FreakMatcher\Eigen/src/Core/../plugins/MatrixCwiseUnaryOps.h:77:32: error: no member named 'binder1st' in namespace 'std'
inline const CwiseUnaryOp<std::binder1st<std::equal_to<Scalar> >, const Derived>
                          ~~~~~^
C:\Users\perda\OneDrive\Documenti\kalwalt-github\webarkit-org\jsartoolkitNFT\emscripten\WebARKitLib\lib\SRC\KPM\FreakMatcher\Eigen/src/Core/../plugins/MatrixCwiseUnaryOps.h:77:64: error: expected '(' for function-style cast or type construction
inline const CwiseUnaryOp<std::binder1st<std::equal_to<Scalar> >, const Derived>
                                         ~~~~~~~~~~~~~~~~~~~~~ ^
C:\Users\perda\OneDrive\Documenti\kalwalt-github\webarkit-org\jsartoolkitNFT\emscripten\WebARKitLib\lib\SRC\KPM\FreakMatcher\Eigen/src/Core/../plugins/MatrixCwiseUnaryOps.h:77:65: error: expected member name or ';' after declaration specifiers
inline const CwiseUnaryOp<std::binder1st<std::equal_to<Scalar> >, const Derived>
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
In file included from C:\Users\perda\OneDrive\Documenti\kalwalt-github\webarkit-org\jsartoolkitNFT\emscripten\WebARKitLib\lib\SRC\KPM\FreakMatcher\facade\visual_database_facade.cpp:37:
In file included from C:\Users\perda\OneDrive\Documenti\kalwalt-github\webarkit-org\jsartoolkitNFT\emscripten\WebARKitLib\lib\SRC\KPM\FreakMatcher\matchers/visual_database-inline.h:44:
In file included from C:\Users\perda\OneDrive\Documenti\kalwalt-github\webarkit-org\jsartoolkitNFT\emscripten\WebARKitLib\lib\SRC\KPM\FreakMatcher\matchers/visual_database.h:44:
In file included from C:\Users\perda\OneDrive\Documenti\kalwalt-github\webarkit-org\jsartoolkitNFT\emscripten\WebARKitLib\lib\SRC\KPM\FreakMatcher\homography_estimation/robust_homography.h:45:
In file included from C:\Users\perda\OneDrive\Documenti\kalwalt-github\webarkit-org\jsartoolkitNFT\emscripten\WebARKitLib\lib\SRC\KPM\FreakMatcher\Eigen/Core:347:
C:\Users\perda\OneDrive\Documenti\kalwalt-github\webarkit-org\jsartoolkitNFT\emscripten\WebARKitLib\lib\SRC\KPM\FreakMatcher\Eigen/src/Core/ArrayBase.h:204:21: error: out-of-line definition of 'operator+=' does not match any declaration in 'ArrayBase<Derived>'
ArrayBase<Derived>::operator+=(const ArrayBase<OtherDerived>& other)
                    ^~~~~~~~
14 errors generated.
emcc: error: 'C:/Users/perda/OneDrive/Documenti/kalwalt-github/emsdk/upstream/bin\clang++.exe -target wasm32-unknown-emscripten -fignore-exceptions -fvisibility=default -mllvm -combiner-global-alias-analysis=false -mllvm -enable-emscripten-sjlj -mllvm -disable-lsr -DEMSCRIPTEN -Werror=implicit-function-declaration -IC:\Users\perda\OneDrive\Documenti\kalwalt-github\emsdk\upstream\emscripten\cache\sysroot\include\SDL --sysroot=C:\Users\perda\OneDrive\Documenti\kalwalt-github\emsdk\upstream\emscripten\cache\sysroot -Xclang -iwithsysroot/include\compat -IC:\Users\perda\OneDrive\Documenti\kalwalt-github\webarkit-org\jsartoolkitNFT\emscripten\WebARKitLib\include -IC:\Users\perda\OneDrive\Documenti\kalwalt-github\webarkit-org\jsartoolkitNFT\build/ -IC:\Users\perda\OneDrive\Documenti\kalwalt-github\webarkit-org\jsartoolkitNFT\emscripten/ -IC:\Users\perda\OneDrive\Documenti\kalwalt-github\webarkit-org\jsartoolkitNFT\emscripten\WebARKitLib\lib\SRC\KPM\FreakMatcher -Oz -D HAVE_NFT -D WITH_FILTERING=1 C:\Users\perda\OneDrive\Documenti\kalwalt-github\webarkit-org\jsartoolkitNFT\emscripten\WebARKitLib\lib\SRC\KPM\FreakMatcher\facade\visual_database_facade.cpp -c -o C:\Users\perda\AppData\Local\Temp\emscripten_temp_v4e18v_3\visual_database_facade_28.o' failed (returned 1)

exec error: 1
```

**Comment by kalwalt** [comment]
this was solved see latest commits,
